src/components/accordion.jsx

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect, useState } from "react";
+import { withStyles } from "@material-ui/core/styles";
+import MuiAccordion from "@material-ui/core/Accordion";
+import MuiAccordionSummary from "@material-ui/core/AccordionSummary";
+import MuiAccordionDetails from "@material-ui/core/AccordionDetails";
+import Typography from "@material-ui/core/Typography";
+
+const Accordion = withStyles({
+  root: {
+    outline: "2px solid white",
+    boxShadow: "none",
+    "&:not(:last-child)": {
+      borderBottom: 0,
+    },
+    "&:before": {
+      display: "none",
+    },
+    "&$expanded": {
+      margin: "auto",
+    },
+  },
+  expanded: {},
+})(MuiAccordion);
+
+const AccordionSummary = withStyles({
+  root: {
+    backgroundColor: "#f7f7f7",
+    marginBottom: -1,
+    minHeight: 56,
+    height: 61,
+
+    "&$expanded": {
+      minHeight: 56,
+      outline: "none",
+      border: "none",
+    },
+  },
+
+  content: {
+    "&$expanded": {
+      margin: "14px 0",
+    },
+  },
+  expanded: {},
+})(MuiAccordionSummary);
+
+const AccordionDetails = withStyles((theme) => ({
+  root: {
+    backgroundColor: "#f7f7f7",
+    padding: theme.spacing(2),
+    paddingTop: theme.spacing(0),
+  },
+}))(MuiAccordionDetails);
+
+export default function CustomizedAccordions() {
+  const [expanded, setExpanded] = useState("panel1");
+
+  const handleChange = (panel) => (event, newExpanded) => {
+    setExpanded(panel);
+  };
+
+  const handleHeaderClassChange = () => {
+    const arrowElementName = `acc-${expanded}-arrow`;
+    console.log(arrowElementName);
+    const arrowElement = document.getElementById(arrowElementName);
+    arrowElement.classList.toggle("accordion-arrow-expanded");
+
+    const headerElementName = `acc-${expanded}-header`;
+    console.log(headerElementName);
+    const headerElement = document.getElementById(headerElementName);
+    headerElement.classList.toggle("accordion-header-expanded");
+  };
+
+  useEffect(() => {
+    console.log(expanded);
+    const arrowElementName = `acc-${expanded}-arrow`;
+    console.log(arrowElementName);
+    const arrowElement = document.getElementById(arrowElementName);
+    arrowElement.classList.toggle("accordion-arrow-expanded");
+
+    const headerElementName = `acc-${expanded}-header`;
+    console.log(headerElementName);
+    const headerElement = document.getElementById(headerElementName);
+    headerElement.classList.toggle("accordion-header-expanded");
+
+    return () => {
+      arrowElement.classList.toggle("accordion-arrow-expanded");
+      headerElement.classList.toggle("accordion-header-expanded");
+    };
+  });
+
+  return (
+    <div className="accordion">
+      <Accordion
+        square
+        expanded={expanded === "panel1"}
+        onChange={handleChange("panel1")}
```

Comment:
why don't you call the hook directly?
like
```
onChange={setExpanded("panel1")}
```

Reply:
When the hook is called directly, an error raised because of too many renders. The function prevents it by a variable called "newExpanded". See below the function pls...

const handleChange = (panel) => (event, newExpanded) => {
    setExpanded(panel);
  };